examples/simple_trainer_mcmc.py

```diff
@@ -433,15 +459,22 @@ def train(self):
                 depthloss = F.l1_loss(disp, disp_gt) * self.scene_scale
                 loss += depthloss * cfg.depth_lambda
 
-            loss = (
-                loss
-                + cfg.opacity_reg
-                * torch.abs(torch.sigmoid(self.splats["opacities"])).mean()
-            )
-            loss = (
-                loss
-                + cfg.scale_reg * torch.abs(torch.exp(self.splats["scales"])).mean()
-            )
+            if cfg.normal_consistency_loss:
+                depths = renders[..., -1:]
+                normals = renders[..., -4:-1]
+                normals_surf = depth_to_normal(
+                    depths,
+                    camtoworlds,
+                    Ks,
+                    near_plane=cfg.near_plane,
+                    far_plane=cfg.far_plane,
+                )
+                normals_surf = normals_surf * (alphas).detach()
```

Comment:
Why do you need to multiply with alphas? Isn’t normals_surf already normalized here: https://github.com/jefequien/gsplat-mcmc/blob/6d156d2d447d9557e432b7f4b94fa6b3b9527187/gsplat/normal_utils.py#L41?

Reply:
You can either normalize both the rendered normals and surface normals or use the alpha-composited rendered normals and multiply the surface normals by alpha. Doing the latter looks better visually.

Reply:
Thanks for the clarification. I was using the first one in GOF.

Reply:
The latter one should be more robust when dealing with semi-transparent pixels (accumulate_alpha < 1).